### ISSUE TITLE
[UWP] Consider using std::vector internally rather than passing an IVector

### ISSUE THREAD

**Issue by paulcam206** [microsoft/AdaptiveCards#3377]
This probably needs a bit more investigation, but it looks like we can improve perf (and legibility) a bit by only dealing with `IVector<>*`s when crossing ABI boundaries. This is [in line](https://docs.microsoft.com/en-us/cpp/cppcx/collections-c-cx?view=vs-2019#vector-usage) with current recommendations:

> If you are passing a sequence within your own program, then use either `Vector` or `std::vector` because they are more efficient than `IVector`. Use `IVector` only when you pass the container across the ABI.

Bonus points, the `Platform::Collections::Vector::{ctor}` can take a [`std::vector` argument](https://docs.microsoft.com/en-us/cpp/cppcx/platform-collections-vector-class?view=vs-2019#ctor). Indeed, the top article linked suggests using `std::vector` is a good way to go:

> If you have existing code that uses `std::vector` and you want to reuse it in a Windows Runtime component, just use one of the `Vector` constructors that takes a `std::vector` or a pair of iterators to construct a `Vector` at the point where you pass the collection across the ABI.

**Comment by paulcam206** [comment]
I talked a bit with @RebeccaAnne, and it looks like we don't have easy access to `Platform::Collections::Vector`. Instead, we have our own internal implementation that uses `std::vector` under the hood. However, I took a look at `Vector.h`, and it looks like we should be able to add the ability to instantiate a `Vector` passing in a `std::vector` to wrap.

https://github.com/microsoft/AdaptiveCards/blob/10dd6f373de16f845539967759454235bf36b9d8/source/uwp/Renderer/lib/Vector.h#L286-L301
